examples/api/lib/widgets/hardware_keyboard/key_event_manager.0.dart

```diff
@@ -0,0 +1,158 @@
+
+  @override
+  State<StatefulWidget> createState() => FallbackDemoState();
+}
+
+class FallbackDemoState extends State<FallbackDemo> {
+  String? _capture;
+  late final FallbackFocusNode _node = FallbackFocusNode(
+    onKeyEvent: (KeyEvent event) {
+      if (event is! KeyDownEvent) {
+        return false;
+      }
+      setState(() {
+        _capture = event.logicalKey.keyLabel;
+      });
+      // TRY THIS: Change the return value to true. You will no longer be able
+      // to type text, because these key events will no longer be sent to the
+      // text input system.
+      return false;
+    }
+  );
+
+  @override
+  Widget build(BuildContext context) {
+    return FallbackFocus(
+      node: _node,
+      child: Container(
+        decoration: BoxDecoration(border: Border.all(color: Colors.red)),
+        padding: const EdgeInsets.all(10),
+        constraints: const BoxConstraints(maxWidth: 500, maxHeight: 400),
+        child: Column(
+          children: <Widget>[
+            const Text('This area handles key pressses that are unhandled by any shortcuts by displaying them below. '
+              'Try text shortcuts such as Ctrl-A!'),
+            Text(_capture == null ? '' : '$_capture is not handled by shortcuts.'),
+            const TextField(decoration: InputDecoration(label: Text('Text field 1'))),
+            Shortcuts(
+              shortcuts: <ShortcutActivator, Intent>{
+                const SingleActivator(LogicalKeyboardKey.keyQ): VoidCallbackIntent(() {}),
+              },
+              child: const TextField(
+                decoration: InputDecoration(label: Text('This field also considers key Q as a shortcut (that does nothing).')),
+              ),
+            ),
+          ],
+        ),
+      )
+    );
+  }
+}
+
+/// A node used by [FallbackKeyEventRegistrar] to register fallback key handlers.
+class FallbackFocusNode {
+  FallbackFocusNode({required this.onKeyEvent});
+
+  final KeyEventCallback onKeyEvent;
+}
+
+/// A global singleton class that allows [FallbackFocus] to register fallback
+/// key event handlers.
+///
+/// This class is initialized when [instance] is first called, at which time it
+/// patches [ServicesBinding.instance.keyEventManager.keyMessageHandler] with
+/// its own handler.
+class FallbackKeyEventRegistrar {
+  FallbackKeyEventRegistrar._();
+  static FallbackKeyEventRegistrar get instance {
+    if (!_initialized) {
+      final KeyMessageHandler? existing = ServicesBinding.instance.keyEventManager.keyMessageHandler;
+      assert(existing != null);
+      ServicesBinding.instance.keyEventManager.keyMessageHandler = _instance._buildHandler(existing!);
+      _initialized = true;
+    }
+    return _instance;
+  }
+  static bool _initialized = false;
+  static final FallbackKeyEventRegistrar _instance = FallbackKeyEventRegistrar._();
+
+  final List<FallbackFocusNode> _fallbackNodes = <FallbackFocusNode>[];
+
+  KeyMessageHandler _buildHandler(KeyMessageHandler existing) {
```

Comment:
This method is also really important. It's probably worth a comment about what it's doing, and why, and to what degree might a reader need/want to alter this implementation.